CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/CVKSwapChain.java

```diff
@@ -649,26 +658,35 @@ public int InitVKDescriptorPool(MemoryStack stack, CVKSynchronizedDescriptorType
                 if (count > 0) {
                     VkDescriptorPoolSize poolSize = pPoolSizes.get(iPoolSize++);
                     poolSize.type(iType);
-                    int size = CalculateDescriptorPoolSizeForType(iType, 
-                                                                  poolDescriptorTypeCounts[iType],
-                                                                  count); 
+                    
+                    // Leaner and meaner version. Just allocate what we need. This could get out of control with 1000s of nodes.
+                    int size = count;
+                    //int size = CalculateDescriptorPoolSizeForType(iType, 
```

Comment:
Why did you neuter that code path?  Sizing the pool to the current size means we'll be recreating the pool every time a new icon is added.  That seems wasteful.  Also I'm not sure what happens to all the existing descriptors when the pool is recreated.  Can you test that (later if fine) on the pool recreation please?

Reply:
restated the old code path